src/Reflection/ReflectionClass.php

```diff
@@ -338,20 +338,33 @@ private function getMethodsIndexedByName() : array
     /**
      * Fetch an array of all methods for this class.
      *
+     * @param ?int $filter
      * @return ReflectionMethod[]
      */
-    public function getMethods() : array
+    public function getMethods(?int $filter = null) : array
     {
-        return array_values($this->getMethodsIndexedByName());
+        if (null === $filter) {
```

Comment:
@roukmoute `is_null()` is a function: while it allows for function composition, calling functions in PHP is the slowest thing ever, and the first/simplest thing you can do to remove `O(1)` overhead on any codebase.

To clarify:

```php
// good
array_keys(array_filter([null, 1, 2, null], 'is_null'));

// bad
if (is_null($something)) {
    // ...
}

// good
if (null === $something) {
    // ...
}
```